// @teuteuf/react-pwa-install/lib/InstallDialog.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _core = require("@material-ui/core");

var _useMediaQuery = _interopRequireDefault(require("@material-ui/core/useMediaQuery"));

var _styles = require("@material-ui/core/styles");

var _InstallDialogAction = _interopRequireDefault(require("./InstallDialogAction"));

var _this = void 0,
    _jsxFileName = "/home/teuteuf/workspace/web/react-pwa-install/src/InstallDialog.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var InstallDialog = function InstallDialog(props) {
  var theme = (0, _styles.useTheme)();
  var fullScreen = (0, _useMediaQuery["default"])(theme.breakpoints.down('xs'));
  return /*#__PURE__*/_react["default"].createElement(_core.Dialog, {
    open: props.open,
    onClose: props.onClose,
    "aria-labelledby": "dialog-title",
    fullScreen: fullScreen,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 12,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react["default"].createElement(_core.DialogTitle, {
    id: "dialog-title",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 7
    }
  }, props.title || "Install Web App"), /*#__PURE__*/_react["default"].createElement(_core.DialogContent, {
    dividers: true,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react["default"].createElement(_core.Grid, {
    container: true,
    alignItems: "center",
    style: {
      height: '100%'
    },
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 9
    }
  }, !!props.logo && /*#__PURE__*/_react["default"].createElement(_core.Grid, {
    item: true,
    xs: 12,
    sm: 6,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 17,
      columnNumber: 13
    }
  }, /*#__PURE__*/_react["default"].createElement(_core.Box, {
    mr: 1,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 18,
      columnNumber: 15
    }
  }, /*#__PURE__*/_react["default"].createElement("img", {
    src: props.logo,
    alt: "logo",
    style: {
      width: '100%'
    },
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 19,
      columnNumber: 17
    }
  }))), !!props.features && /*#__PURE__*/_react["default"].createElement(_core.Grid, {
    item: true,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 24,
      columnNumber: 13
    }
  }, /*#__PURE__*/_react["default"].createElement(_core.Box, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 25,
      columnNumber: 15
    }
  }, /*#__PURE__*/_react["default"].createElement(_core.Typography, {
    variant: "subtitle1",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 17
    }
  }, props.featuresTitle || "Key Features:"), /*#__PURE__*/_react["default"].createElement(_core.Typography, {
    variant: "body2",
    component: "div",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 17
    }
  }, props.features))), !!props.description && /*#__PURE__*/_react["default"].createElement(_core.Grid, {
    item: true,
    xs: 12,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 34,
      columnNumber: 13
    }
  }, /*#__PURE__*/_react["default"].createElement(_core.Typography, {
    variant: "subtitle1",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 35,
      columnNumber: 15
    }
  }, props.descritpionTitle || "Description:"), /*#__PURE__*/_react["default"].createElement(_core.Typography, {
    variant: "body2",
    component: "div",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 36,
      columnNumber: 15
    }
  }, props.description)))), /*#__PURE__*/_react["default"].createElement(_core.DialogContent, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react["default"].createElement(_InstallDialogAction["default"], {
    platform: props.platform,
    onSubmit: props.onSubmit,
    onClose: props.onClose,
    instructionTitle: props.instructionTitle,
    instructionActionOk: props.instructionActionOk,
    instructionActionCancel: props.instructionActionCancel,
    instructionActionInstall: props.instructionActionInstall,
    instructionIdeviceAction1: props.instructionIdeviceAction1,
    instructionIdeviceAction2: props.instructionIdeviceAction2,
    instructionFirefoxAction1: props.instructionFirefoxAction1,
    instructionFirefoxAction2: props.instructionFirefoxAction2,
    instructionFirefoxNewAction1: props.instructionFirefoxNewAction1,
    instructionFirefoxNewAction2: props.instructionFirefoxNewAction2,
    instructionOperaAction1: props.instructionOperaAction1,
    instructionOperaAction2: props.instructionOperaAction2,
    instructionNotSupported: props.instructionNotSupported,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 44,
      columnNumber: 9
    }
  })));
};

var _default = InstallDialog;
exports["default"] = _default;
module.exports = exports.default;
